docs/docs/community/release-process.md

```diff
@@ -0,0 +1,75 @@
+---
+id: release-process
+title: The Nebari release process
+description: The Nebari release process and workflow
+---
+
+# The Nebari release process
+
+This page describes the high-level details of cutting a new Nebari release.
+
+|                   |                                                                                                                                                                                                  |
+| :---------------: | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
+|      Cadence      |                                                                                             Monthly                                                                                              |
+| Versioning System |                                                                                    [CalVer](#calver-details)                                                                                     |
+| Release Checklist |     [Link to issue template](https://github.com/nebari-dev/nebari/issues/new?assignees=&labels=type%3A+release+%F0%9F%8F%B7&template=release-checklist.md&title=%5BRELEASE%5D+%3Cversion%3E)     |
+| Testing Checklist | [Link to issue template](https://github.com/nebari-dev/nebari/issues/new?assignees=&labels=type%3A+release+%F0%9F%8F%B7&template=testing-checklist.md&title=Testing+checklist+for+%3Cversion%3E) |
+
+## Release Captain responsibilities
+
+For every release, there is an assigned "Release Captain". It is the responsibility of this person to manage both the release and testing checklists as well as to communicate the status of the release to the rest of the Nebari development team and the community.
+
+Communication channels that need to be updated are:
+
+- the release checklist issue itself
+- the `#nebari-dev` slack channel
+
+## CalVer details
+
+These releases should follow the following CalVer versioning style:
+
+```
+YYYY-MM-releaseNumber
+```
+
+:::info
+`YYYY` represents the year - i.e. `2023`
+
+`MM` represents the _non-zero padded_ month - i.e. `1` for January, `12` for December
+
+`releaseNumber` represents the current release for that month, starting at `1`. Anything above `1` represents a [hotfix (patch) release](#hotfixes).
+:::
+
+:::caution
+For the release tag, there should be NO prepended `v`
+:::
+
+For example, the first Nebari CalVer release was `2022.10.1`. If a hotfix release was needed in the same month, we simply increment the `releaseNumber` by 1, which would be `2022.10.2` (_this is to illustrate how the increment works, this release does not exist._)
+
+## Gitflow and branching strategy
```

Comment:
I know this is somewhat related to releases, but I would prefer having the gitflow + branching in a separate doc. 

As a newcomer, I would not look at the release process if I only care about how to open a PR and which branch to use.

Reply:
As you mentioned, these concepts are closely linked and I believe they should remain together. We reference the "release process" quite a bit in the "Gitflow" section and if the our branching strategy makes sense, newcomers shouldn't need to decide which branch they need to open their PR against 😊

Reply:
I'd also like to make the branching strategy more visible/finable in the sidebar, for folks who might be looking exactly for this information. Bokeh contributors tend to look for these details when they notice the branch names because it is not a super common workflow.

If we decide to not create a separate doc, I think it'll be worth including it in the page title: "Release process and branching strategy". :)

Reply:
Much better